ddtrace/internal/ci_visibility/recorder.py

```diff
@@ -67,6 +71,35 @@ def __init__(self, tracer=None, config=None, service=None):
         except Exception:
             log.warning("Failed to load CODEOWNERS", exc_info=True)
 
+    def _configure_writer(self):
+        writer = None
+        if ddconfig._ci_visibility_agentless_enabled:
+            site_id = os.environ.get("DD_SITE")
+            if site_id:
+                writer = CIVisibilityWriter(intake_url="https://citestcycle-intake.%s" % site_id)
+            else:
+                log.error("Environment variable DD_SITE not set - not sending CI Visibility data")
```

Comment:
If `DD_API_KEY` isn't set, according to https://github.com/DataDog/dd-trace-py/pull/5497#discussion_r1166445998 we shouldn't report anything right? Should we raise an exception here instead?

Reply:
This is another area of ambiguity. @juan-fernandez if `DD_API_KEY` is unset and `DD_CIVISIBILITY_AGENTLESS_ENABLED` is set, should ddtrace

* crash with an error message; or
* generate a warning message and let the test suite continue to run, sending non ci-visibility spans from the pytest integration to the datadog agent

Reply:
other tracers log an error message (or warning, depending on the language) and continue the test execution _without any tracing or ci visibility_, that is, the tracer is not init. Example: https://github.com/DataDog/dd-trace-js/blob/master/ci/init.js#L27-L30

Reply:
Sounds good, thank you. I've adjusted the diff to do the same.

Reply:
my last comment is wrong, my bad. Javascript is the odd one here. Let's fail the test process if `DD_CIVISIBILITY_AGENTLESS_ENABLED` is set and `DD_API_KEY` is unset.